superset-frontend/src/components/TableSelector/index.tsx

```diff
@@ -174,56 +185,73 @@ const TableSelector: FunctionComponent<TableSelectorProps> = ({
     // reset selections
     if (database === undefined) {
       setCurrentDatabase(undefined);
+      setCurrentCatalog(undefined);
       setCurrentSchema(undefined);
       setCurrentTable(undefined);
     }
   }, [database]);
 
   useEffect(() => {
-    if (currentDatabase && currentSchema) {
+    if (shouldLoadTables(currentDatabase)) {
       setLoadingTables(true);
-      const encodedSchema = encodeURIComponent(currentSchema);
       const forceRefresh = refresh !== previousRefresh;
-      // TODO: Would be nice to add pagination in a follow-up. Needs endpoint changes.
```

Comment:
Would be nice to retain the TODO my moving it into the `getEndpoint` callback

Reply:
Moved to `getEndpoint` callback